### ISSUE TITLE
JAX-RS parameters are all required by default

### ISSUE THREAD

**Issue by neon-dev** [smallrye/smallrye-open-api#130]
The following code produces below openapi file
```java
@Path("test")
public class TestRS {

  @POST
  @Path("")
  public void test(@CookieParam ("cookieParam") String c, @HeaderParam("headerParam") String h, @QueryParam("queryParam") String q) {
  }
}
```

```yml
---
openapi: 3.0.1
info:
  title: Generated API
  version: "1.0"
paths:
  /rest/test:
    post:
      parameters:
      - name: cookieParam
        in: cookie
        required: true
        schema:
          type: string
      - name: headerParam
        in: header
        required: true
        schema:
          type: string
      - name: queryParam
        in: query
        required: true
        schema:
          type: string
      responses:
        201:
          description: Created
```

According to https://github.com/eclipse/microprofile-open-api/issues/355#issuecomment-496910929 I'd expect only @PathParam of JAX-RS to be automatically set as required. The parameter types in the example above should have no required property or with a value of `false`.

I couldn't find the reason for the current behavior while jumping through the source code on GitHub, because at least here it looks fine:
https://github.com/smallrye/smallrye-open-api/blob/e29bdabc89aae81e00be32cdb4dd5a34f551f6c2/implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java#L555

**Comment by EricWittmann** [comment]
Thanks for the report - I'm sure it's a quick fix (he said hopefully). :)

**Comment by MikeEdgar** [comment]
@EricWittmann , @neon-dev - I think this is a timing issue. The issue has been fixed in master, but there has not yet been a release deployed to Maven central.

**Comment by EricWittmann** [comment]
See - I told you it would be a quick fix!  :)

Hey @kenfinnigan and @msavy - any objection to doing a release of this library?

**Comment by EricWittmann** [comment]
We've got one more PR that needs to be reviewed first, I think.

**Comment by kenfinnigan** [comment]
Was planning a release with config update, so we can wait for another PR

**Comment by EricWittmann** [comment]
OK I'm just waiting on the completion of the discussion happening in https://github.com/smallrye/smallrye-open-api/issues/125 before I merge.

**Comment by EricWittmann** [comment]
@kenfinnigan OK we're good to go for a new release. :)  At your convenience.
